### PR TITLE
refactor(ivy): remove instruction usage from other instructions

### PR DESCRIPTION
**Pull request by crisbeto**
Removes direct calls from one instruction into another, moves the shared logic into a separate function and removes the state getters from the shared function.

This PR resolves FW-1340.

### CODE REVIEW

File: packages/core/src/render3/instructions/styling.ts [hunk 1]
```diff
@@ -146,9 +146,15 @@ function initStyling(
...
+}
+
+export function stylePropInternal(
+    lView: LView, selectedIndex: number, styleIndex: number,
+    value: string | number | String | PlayerFactory | null, suffix?: string | null,
+    forceOverride?: boolean): void {
   const valueToAdd = resolveStylePropValue(value, suffix);
-  const stylingContext = getStylingContext(index, getLView());
+  const stylingContext = getStylingContext(selectedIndex, lView);
   const directiveStylingIndex = getActiveDirectiveStylingIndex();
```
Comment (kara) [inline packages/core/src/render3/instructions/styling.ts]: Can we move this call up to the instruction too? It just combines two global vars. Same in `classMapInternal`

File: packages/core/src/render3/instructions/listener.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export function ɵɵlistener(
  *
  * @codeGenApi
 */
-export function ɵɵcomponentHostSyntheticListener<T>(
+export function ɵɵcomponentHostSyntheticListener(
```
Comment (kara) [inline packages/core/src/render3/instructions/listener.ts]: This change seems somewhat unrelated to the others. Why is it necessary?

**Review by kara — APPROVED** [review]
Thanks for this cleanup! Some nits below

File: packages/core/src/render3/instructions/listener.ts (hunk 2 above)
Comment (crisbeto) [reply packages/core/src/render3/instructions/listener.ts]: I noticed that the `T` wasn't being used while I was scanning through the file so I removed it.

**Comment by crisbeto** [comment]
The feedback has been addressed @kara.

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
[VE presubmit](https://test.corp.google.com/ui#id=OCL:257420223:BASE:257420248:1562776362707:9f9f8560)

[Ivy presubmit](http://test/OCL:257420223:BASE:257420742:1562776536329:79c60cd0)

**Comment by kara** [comment]
merge-assistance: codefresh is red on master + global approval
